test/parallel/test-assert-objects.js

```diff
@@ -0,0 +1,399 @@
+    const sym = Symbol('test');
+    const obj1 = { num: 1, str: 'test', bool: true, sym: sym };
+    const obj2 = { num: 1, str: 'test', bool: true, sym: sym };
+    assert.matchObjectStrict(obj1, obj2);
+  });
+
+  it('should compare two objects with Buffers', function() {
+    const obj1 = { buf: Buffer.from('Node.js') };
+    const obj2 = { buf: Buffer.from('Node.js') };
+    assert.matchObjectStrict(obj1, obj2);
+    assert.matchObject(obj1, obj2);
+  });
+
+  it('should strictly compare two objects with identical Error properties', function() {
+    const error = new Error('Test error');
+    const obj1 = { error: error };
+    const obj2 = { error: error };
+    assert.matchObjectStrict(obj1, obj2);
+  });
+
+  it('should not strictly compare two objects with different Error instances', function() {
+    const obj1 = { error: new Error('Test error 1') };
+    const obj2 = { error: new Error('Test error 2') };
+    assert.throws(() => assert.matchObjectStrict(obj1, obj2), Error);
+  });
+
+  it('should strictly compare two objects with the same TypedArray instance', function() {
+    const typedArray = new Uint8Array([1, 2, 3]);
+    const obj1 = { typedArray: typedArray };
+    const obj2 = { typedArray: typedArray };
+    assert.matchObjectStrict(obj1, obj2);
+  });
+
+  it('should not strictly compare two objects with different TypedArray instances and content', function() {
+    const obj1 = { typedArray: new Uint8Array([1, 2, 3]) };
+    const obj2 = { typedArray: new Uint8Array([4, 5, 6]) };
+    assert.throws(() => assert.matchObjectStrict(obj1, obj2), Error);
+  });
+
+  it('should not strictly compare two objects with different TypedArray instances', function() {
+    const obj1 = { typedArray: new Uint8Array([1, 2, 3]) };
+    const obj2 = { typedArray: new Uint8Array([1, 2, 3]) };
+    assert.matchObjectStrict(obj1, obj2);
+  });
+
+  it('should strictly compare two Map objects with identical entries', function() {
+    const map1 = new Map([
+      ['key1', 'value1'],
+      ['key2', 'value2'],
+    ]);
+    const map2 = new Map([
+      ['key1', 'value1'],
+      ['key2', 'value2'],
+    ]);
+    assert.matchObjectStrict(map1, map2);
+  });
+
+  it('should not strictly compare two Map objects with different entries', function() {
+    const map1 = new Map([
+      ['key1', 'value1'],
+      ['key2', 'value2'],
+    ]);
+    const map2 = new Map([
+      ['key1', 'value1'],
+      ['key3', 'value3'],
+    ]);
+    assert.throws(() => assert.matchObjectStrict(map1, map2), Error);
+  });
+
+  it('should strictly compare two Set objects with identical values', function() {
+    const set1 = new Set(['value1', 'value2']);
+    const set2 = new Set(['value1', 'value2']);
+    assert.matchObjectStrict(set1, set2);
+  });
+
+  it('should strictly compare two Map objects from different realms with identical entries', function() {
+    const map1 = new vm.runInNewContext('new Map([["key1", "value1"], ["key2", "value2"]])');
```

Comment:
nit: we can get more syntax highlighting and lint checks by getting only the constructor from the other realm and calling it locally
```suggestion
    const map1 = new vm.runInNewContext('Map')([["key1", "value1"], ["key2", "value2"]]));
```

Reply:
Doing it like that it throws an error `TypeError: Constructor Map requires 'new'`, same for the Set below

Reply:
Ah yes, we need extra parentheses, my bad
```suggestion
    const map1 = new (vm.runInNewContext('Map'))([["key1", "value1"], ["key2", "value2"]]);
```